web/shapeworks/src/components/ShapeViewer/imageSync.js

```diff
@@ -0,0 +1,380 @@
+                    ) {
+                        visible[i] = 1
+                    }
+                }
+
+                arraysOut.forEach(x => x.modified())
+            }
+        })
+        filteredParticleMapper.setScaleModeToScaleByMagnitude();
+        filteredParticleMapper.setScaleFactor(this.glyphSize);
+        filteredParticleMapper.setScaleArray('visible')
+        filteredParticleMapper.setInputConnection(particleFilter.getOutputPort(), 0)
+        filteredParticleMapper.setInputConnection(filteredParticleSphere.getOutputPort(), 1)
+        filteredParticleActor.setMapper(filteredParticleMapper);
+
+        const slice = {
+            renderer,
+            sliceActor,
+            imageMapper,
+            cutPlane,
+            cutter,
+            cutMapper,
+            cutActor,
+            cropFilter,
+            particleFilter,
+            filteredParticleMapper,
+            filteredParticleActor,
+        }
+        slices.value.push(slice)
+        return slice
+    },
+    updateSlice(slice) {
+        const { sliceActor, imageMapper, renderer } = slice;
+
+        let cameraPosition, cameraDirection, cameraViewUp;
+        const camera = renderer.getActiveCamera();
+        const center = sliceActor.getCenter();
+        const distance = camera.getDistance();
+
+        if (['X', 'L'].includes(imageViewAxis.value)) {
+            imageMapper.setSlicingMode('I');
+            imageMapper.setISlice(imageViewSlices.value.x);
+            cameraPosition = [
+                center[0] + distance,
+                center[1],
+                center[2],
+            ]
+            cameraDirection = [-1, 0, 0]
+            cameraViewUp = [0, 1, 0]
+        } else if (['Y', 'P'].includes(imageViewAxis.value)) {
+            imageMapper.setSlicingMode('J');
+            imageMapper.setJSlice(imageViewSlices.value.y);
+            cameraPosition = [
+                center[0],
+                center[1] + distance,
+                center[2],
+            ]
+            cameraDirection = [0, -1, 0]
+            cameraViewUp = [0, 0, -1]
+        } else if (['Z', 'S'].includes(imageViewAxis.value)) {
+            imageMapper.setSlicingMode('K');
+            imageMapper.setKSlice(imageViewSlices.value.z);
+            cameraPosition = [
+                center[0],
+                center[1],
+                center[2] + distance,
+            ]
+            cameraDirection = [0, 0, -1]
+            cameraViewUp = [0, 1, 0]
+        }
+
+        if (cameraPosition && cameraDirection && cameraViewUp) {
+            camera.setPosition(...cameraPosition)
+            camera.setDirectionOfProjection(...cameraDirection)
+            camera.setViewUp(...cameraViewUp)
+        }
+
+        if (imageViewIntersectMode.value) {
+            this.updateCuttingPlane(slice)
+        }
+        renderer.resetCamera()
+    },
+    updateIntersection(slice) {
+        // TODO: account for multidomain cases
```

Comment:
Is there a plan in place for this? The image slice should be the same regardless of domain number.

Reply:
I will need a dataset to use to implement this; I don't currently have a multidomain dataset where the original data are images. The slices will be the same, but there would be `n` shapes to intersect and `n` particles objects to filter.